src/level_four/core/tasks/send_transaction_tasks_solana.c

```diff
@@ -0,0 +1,225 @@
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+#include "btc.h"
+#include "constant_texts.h"
+#include "contracts.h"
+#include "controller_level_four.h"
+#include "int-util.h"
+#include "math.h"
+#include "solana.h"
+#include "tasks_level_four.h"
+#include "tasks_tap_cards.h"
+#include "ui_address.h"
+#include "ui_confirmation.h"
+#include "ui_delay.h"
+#include "ui_input_text.h"
+#include "ui_instruction.h"
+#include "ui_message.h"
+#include "utils.h"
+
+extern char *ALPHABET;
+extern char *ALPHA_NUMERIC;
+extern char *NUMBERS;
+extern char *PASSPHRASE;
+
+extern lv_task_t *timeout_task;
+
+void send_transaction_tasks_solana() {
+  switch (flow_level.level_three) {
+    case SEND_TXN_VERIFY_COIN_SOLANA: {
+      instruction_scr_init(ui_text_processing, NULL);
+      mark_event_over();
+    } break;
+
+    case SEND_TXN_UNSIGNED_TXN_WAIT_SCREEN_SOLANA: {
+      mark_event_over();
+    } break;
+
+    case SEND_TXN_UNSIGNED_TXN_RECEIVED_SOLANA: {
+      mark_event_over();
+    } break;
+
+    case SEND_TXN_VERIFY_CONTRACT_ADDRESS: {
+      mark_event_over();
+    } break;
+
+    case SEND_TXN_VERIFY_TXN_NONCE_SOLANA: {
+      mark_event_over();
+    } break;
+
+    case SEND_TXN_VERIFY_RECEIPT_ADDRESS_SOLANA: {
+      instruction_scr_destructor();
+      char address[45]      = {0};
+      char top_heading[225] = {0};
+      char display[70]      = {0};
+
+      instruction_scr_destructor();
+      size_t address_size = sizeof(address);
+      b58enc(address, &address_size, solana_unsigned_txn_ptr.instruction.program.transfer.recipient_account,
+             SOLANA_ACCOUNT_ADDRESS_LENGTH);
+      snprintf(top_heading, sizeof(top_heading), "%s", ui_text_verify_address);
+      snprintf(display, sizeof(display), "%s%s", ui_text_20_spaces, address);
+      address_scr_init(top_heading, display, true);
+    } break;
+
+    case SEND_TXN_CALCULATE_AMOUNT_SOLANA: {
+      instruction_scr_init(ui_text_fetching_recipient_amount, NULL);
```

Comment:
drop this. [refer here](https://github.com/Cypherock/x1_wallet_firmware/pull/30/files#diff-162eb16d5f68d2055ea046c1b5ca28032abd6cd16e23f69435236e1798f16da0R158-R159)